### ISSUE TITLE
presentFullscreenPlayer() is throwing an error "Application tried to present modal view controller on itself"

### ISSUE THREAD

**Issue by pfinazzo** [TheWidlarzGroup/react-native-video#3029]
# Bug

## Platform

Which player are you experiencing the problem on:
* iOS

- Not sure about other devices

## Environment info

I could not run react-native-info, not sure why. But here is my package.json

I used yarn to add the package to node_modules and performed a `pod install` in my ios directory. 

Machine info:
<img width="213" alt="Screen Shot 2023-02-08 at 5 39 56 PM" src="https://user-images.githubusercontent.com/29343873/217667533-e2f3fdb9-2625-45fa-81c7-3c54e01ccf42.png">

```
{
  "name": "tonebase",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "npx react-native run-android",
    "ios": "npx react-native run-ios",
    "start": "react-native start",
    "ios-clean-start": "npm run total-cleanup && kill-port 8081 && npm run ios",
    "total-cleanup": "npm run cleanup-nodemodules && npm run cleanup-ios",
    "cleanup-nodemodules": "rm -rf node_modules && rm package-lock.json && npm i",
    "cleanup-ios": "cd ios && rm -rf Pods && rm Podfile.lock && pod install && cd ..",
    "clean-android": "rm -rf android/build && rm -rf android/app/build && mkdir android/app/build && mkdir android/build && mkdir android/app/build/outputs",
    "clean-ios": "rm -rf ios/build && mkdir ios/build",
    "clean": "yarn run clean-android && yarn run clean-ios",
    "test": "jest",
    "lint": "eslint .",
    "beta:android": "cd ./android && bundle install && bundle exec fastlane android beta version:patch",
    "beta:ios": "cd ./ios && bundle install && bundle exec fastlane ios beta version:patch",
    "beta": "yarn run beta:android && yarn run beta:ios",
    "release:android": "cd ./android && bundle exec fastlane android release version:patch",
    "release:ios": "cd ./ios && bundle exec fastlane ios release version:patch",
    "release": "yarn run release:android && yarn run release:ios",
    "register-device": "cd ios && fastlane devices && cd .."
  },
  "dependencies": {
    "@miblanchard/react-native-slider": "^2.1.0",
    "@react-native-async-storage/async-storage": "^1.17.10",
    "@react-native-community/checkbox": "^0.5.14",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/slider": "^4.3.1",
    "@react-native-firebase/app": "^15.7.0",
    "@react-native-firebase/auth": "^15.7.0",
    "@react-native-firebase/messaging": "^15.7.0",
    "@react-native-google-signin/google-signin": "^8.0.1",
    "@react-navigation/bottom-tabs": "^6.4.0",
    "@react-navigation/native": "^6.0.13",
    "@react-navigation/native-stack": "^6.9.1",
    "@react-navigation/stack": "^6.3.2",
    "@risingstack/react-easy-state": "^6.3.0",
    "axios": "^0.27.2",
    "dayjs": "^1.11.6",
    "lodash": "^4.17.21",
    "react": "18.1.0",
    "react-native": "0.70.1",
    "react-native-bouncy-checkbox": "^3.0.6",
    "react-native-gesture-handler": "^2.7.0",
    "react-native-inappbrowser-reborn": "^3.7.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-linear-gradient": "^2.6.2",
    "react-native-paper": "^5.0.0-rc.6",
    "react-native-permissions": "^3.6.1",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "^4.4.1",
    "react-native-screens": "^3.18.2",
    "react-native-svg": "^13.2.0",
    "react-native-toast-message": "^2.1.5",
    "react-native-video": "^6.0.0-alpha.4",
    "react-native-vimeo-iframe": "^1.2.0",
    "react-native-webview": "^11.26.0",
    "react-navigation-shared-element": "^3.1.3",
    "string-similarity": "^4.0.4"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-jest": "^26.6.3",
    "eslint": "^7.32.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.72.1",
    "react-native-sass-transformer": "^2.0.0",
    "react-test-renderer": "18.1.0",
    "sass": "^1.55.0"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

Library version: ^6.0.0-alpha.4

## Steps To Reproduce

1. Try to open fullscreen player via `videoPlayerInstance?.current?.presentFullscreenPlayer()` on iOS emulator

...

## Expected behaviour

1. It opens full screen properly
2.

## Screenshot

<img width="434" alt="Screen Shot 2023-02-08 at 5 38 38 PM" src="https://user-images.githubusercontent.com/29343873/217667351-1ac62b83-1ccf-4ffa-98ec-c1b7b34f2513.png">

**Comment by pfinazzo** [comment]
wanted to also say this was working a few days ago, and not sure if its only not working in emulator or in test flight etc

**Comment by freeboub** [comment]
Can you detail which props are passed to react native video please ?

**Comment by freeboub** [comment]
Is this a duplicated of https://github.com/react-native-video/react-native-video/issues/3025 ?

**Comment by pfinazzo** [comment]
<img width="857" alt="Screen Shot 2023-02-20 at 6 25 31 PM" src="https://user-images.githubusercontent.com/29343873/220212407-c847a3ce-7891-4716-8784-e6d5e118ce61.png">

I dont think so. I don't have controls={true} 

@freeboub

**Comment by pfinazzo** [comment]
It stopped working in both the simulator and in test flight

**Comment by pfinazzo** [comment]
downgrading to 5.2.1 fixes the issue

**Comment by fontesrp** [comment]
Same error on 6.0.0-alpha.5

**Comment by MikeAlvarado** [comment]
**Currently working with** 
```
    "react-native": "0.68.2",
    "react-native-video": "^6.0.0-alpha.6"
```

**Running on Android API 33 & iOS 16.4.1**

To provide more information as I am experiencing the same issue. 
There are some miss behaviors while trying to present fullscreen depending on the platform. 

I have custom controls as an overlay on a View. 

iOS seems to works fine and present fullscreen if `controls={false}` 
While Android works exactly the other way around. fullscreen only works if `controls={true}` and click on the custom control. 

Please see the video on Android. 
(First I present `controls={false}` and try to click my custom control : nothing happens, then I change it to `controls={true}` and with or without the custom control, works as expected. 

https://github.com/react-native-video/react-native-video/assets/9059670/d85bcaf0-678f-4952-8cc3-0d052436dc14

O the other hand, iOS, as stated here, breaks if `controls={false}` and prompt fullscreen, while it works if `controls={true}`. 

https://github.com/react-native-video/react-native-video/assets/9059670/9105fd33-74e2-49e1-969e-1146f235216f

**Comment by MikeAlvarado** [comment]
@freeboub, please let me know if I can upload anything else.

**Comment by nicolaosm** [comment]
This issue is still there and has not been fixed in **v6 alpha7**
@freeboub I believe this is critical issue for final release of v6.0

**Comment by freeboub** [comment]
I just open PR which shall fix the issue here : https://github.com/react-native-video/react-native-video/pull/3230
Can you please test it on your side ?

**Comment by freeboub** [comment]
fix will be integrated in 6.0.0-alpha.8

**Comment by nicolaosm** [comment]
@freeboub Issue is resolved on iOS but on Android the presentFullscreenPlayer() is still broken. Will create a separate issue for it.
